src/MongoDB/Session.c

```diff
@@ -93,6 +93,25 @@ static bool php_phongo_session_get_timestamp_parts(zval* obj, uint32_t* timestam
 	return retval;
 }
 
+static const char* php_phongo_get_transaction_state_string(mongoc_transaction_state_t state)
+{
+	switch (state) {
+		case MONGOC_TRANSACTION_NONE:
+			return "none";
+		case MONGOC_TRANSACTION_STARTING:
+			return "starting";
+		case MONGOC_TRANSACTION_IN_PROGRESS:
+			return "in_progress";
+		case MONGOC_TRANSACTION_COMMITTED:
+			return "committed";
+		case MONGOC_TRANSACTION_ABORTED:
+			return "aborted";
+		default:
+			phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Invalid transaction state %d given", (int) state);
```

Comment:
Considering that libmongoc exposes transaction states as defined in the spec, I've decided to throw an exception in this case. While this requires us to pay close attention to any new transaction states, I'm not sure what other alternative would be feasible (besides returning the integer value)

Reply:
I think an exception is fine here. Based on how [CDRIVER-3364](https://jira.mongodb.org/browse/CDRIVER-3364) was implemented, we know that libmongoc would abort before we ever reach this point. Also, any new transaction state would likely entail a spec change, which would trickle down as CDRIVER and PHPC tickets, so I'm not worried about missing such a change.